department-of-veterans-affairs/caseflow
# Queue | Attorney check out | Remand reasons

Resolves #4484 

### Description
Please explain the changes you made here.

[Remand reasons code mapping](https://github.com/department-of-veterans-affairs/caseflow/issues/4690#issuecomment-373727094)

### Acceptance Criteria
- [x] Verify that if a user selects "Remand" as an issue disposition on the Select Remands Screen #4471, they are directed to the "Select Remands Screen" 
- [x] Verify that regardless of number of remand dispositions selected, one issue is shown
- [x] Verify that each issue is separated by a title called Issue 1, Issue 2 etc. 
- [x] Verify that underneath the Issue title is the issue with the following structure: 
     - Program
     - Issue 
     - Code
     - Certification date 
- [x] Verify that there should be a a sub title caused Remand Reasons 
- [x] Verify that for each remand reason has the following sections as radio buttons in this order:
    - Medical Examinations and Opinions
        - Medical Opinions
        - Medical Examinations
    - Duty to assist 
        - VA records
        - Private records
        - Service personnel records
        - Service treatment records
        - Other government records
     - Due process and duty to notify 
         - Due process
          - Duty to notify 
- [x] Verify that one of the reasons are picked an indented check box appears with the following options
      - Before certification
      - After certification
- [x] Verify that when a user clicks show more 2 more issues appear
- [x] Verify that when the last remand disposition is displayed, the show more disappears 

### Testing Plan
1. Go to the Draft Decision checkout flow, set 1 or more issue dispositions to Remanded, and click to the next page.
2. Verify that choosing any remand reason displays before/after certification radio buttons for that reason. Verify that clicking "Show more" at the bottom renders 2 more remand reason blocks (if applicable).
3. Verify that remand reasons are saved to state after leaving this page (either by going backwards or forwards), and that navigating back to this page restores selected remand reasons.
4. Verify that remand reasons are reset on change issue disposition: set issue to Remanded, set reasons, change disposition, change disposition back to Remanded, verify no reasons are checked.
4. Verify that submitting Draft Decision is successful.

#### Known Issues
1. When selecting issue dispositions, after changing one disposition to "Remanded", the next step button will say "Finish dispositions", rather than "Select remand reasons". If you change any disposition after changing one to "Remanded", the button text will update (#5092)
